src/Nethermind/Nethermind.Evm/EvmObjectFormat.cs

```diff
@@ -0,0 +1,208 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Nethermind.Core.Attributes;
+using Nethermind.Core.Extensions;
+using Nethermind.Evm.CodeAnalysis;
+using Nethermind.Logging;
+using Org.BouncyCastle.Crypto.Paddings;
+
+namespace Nethermind.Evm
+{
+    enum SectionDividor : byte
+    {
+        Terminator = 0,
+        CodeSection = 1,
+        DataSection = 2,
+    }
+    public class EofHeader
+    {
+        #region public construction properties
+        public UInt16 CodeSize { get; set; }
+        public UInt16 DataSize { get; set; }
+        #endregion
+
+        #region Equality methods
+        public override bool Equals(object? obj)
+            => this.GetHashCode() == obj.GetHashCode();
+        public override int GetHashCode()
+            => CodeSize.GetHashCode() ^ DataSize.GetHashCode();
+        #endregion
+
+        #region Sections Offsets
+        private int? _codeStartOffset = default;
+        public int CodeStartOffset
+        {
+            get
+            {
+                if (_codeStartOffset is null)
+                    _codeStartOffset = DataSize == 0 ? 7 : 10;
```

Comment:
Magic numbers 7 : 10? It is hard to understand the context

Reply:
Added comments explaining them : 
```csharp
public int CodeStartOffset
{
    get 
    {
        if (_codeStartOffset is null)
            _codeStartOffset = DataSize == 0
               ? 7     // MagicLength + Version + 1 * (SectionSeparator + SectionSize) + HeaderTerminator = 2 + 1 + 1 * (1 + 2) + 1 = 7
               : 10;   // MagicLength + Version + 2 * (SectionSeparator + SectionSize) + HeaderTerminator = 2 + 1 + 2 * (1 + 2) + 1 = 10 
       return _codeStartOffset.Value;
    }
}
```